plugins/modules/smart_proxy.py

```diff
@@ -101,13 +111,55 @@ def main():
         foreman_spec=dict(
             name=dict(required=True),
             url=dict(required=True),
+            lifecycle_environments=dict(required=False, type='entity_list'),
             download_policy=dict(required=False, choices=['background', 'immediate', 'on_demand']),
         ),
-        required_plugins=[('katello', ['download_policy'])],
+        required_plugins=[('katello', ['lifecycle_environments', 'download_policy'])],
     )
 
     with module.api_connection():
-        module.run()
+        handle_lifecycle_environments = not module.desired_absent and 'lifecycle_environments' in module.foreman_params
+        if handle_lifecycle_environments:
+            module.lookup_entity('lifecycle_environments')
+            lifecycle_environments = module.foreman_params.pop('lifecycle_environments', [])
+
+        smart_proxy = module.lookup_entity('entity')
+        new_smart_proxy = module.run()
+
+        if handle_lifecycle_environments:
+
+            if smart_proxy:
+                payload = {
+                    'id': new_smart_proxy['id'],
+                }
+                current_lces = module.resource_action('capsule_content', 'lifecycle_environments', payload, ignore_check_mode=True, record_change=False)
+            else:
+                current_lces = {'results': []}
```

Comment:
I just realized, this is not only for check_mode now, but also skipping the extra call if we just created a new entity and, therefore, know the result will be empty.
:1st_place_medal: